Yoast/wordpress-seo
# Adapt findKeywordInFirstParagraph for Japanese

## Context


*

## Summary


This PR can be summarized in the following changelog entry:

* [yoastseo] Adds the option to pass a custom `matchWordInText` helper and a helper for matching Japanese keyphrases in the `findKeywordInFirstParagraph` research.

## Relevant technical choices:

* 

## Test instructions for Lingo team

This PR can be acceptance tested by following these steps (**content analysis app**):
- Enable the Japanese morphology data in `getMorphologyData.js` by replacing this line:
`import { isFeatureEnabled } from "@yoast/feature-flag`
 with the followings: 
`import { isFeatureEnabled, enableFeatures } from "@yoast/feature-flag";
enableFeatures( [ "JAPANESE_SUPPORT" ] );`
- Enable sentences segmentation by doing the same in `getSentences.js`
- Set your locale to`ja`
- Create a post
- Set `自然` as the focus keyphrase
-  Add this text to the first paragraph `自然とは、人為によってではなく、おのずから存在しているもの。`
-  Confirm that the Keyphrase in first paragraph assessment returns a green bullet.

**In Docker:**
- Enable the `JAPANESE_SUPPORT` feature by adding `define( 'YOAST_SEO_JAPANESE_SUPPORT', true );` in `basic-wordpress-config.php` file
- Set your site to Japanese` 日本語`
- Create a post
- Set `自然の中を歩く` as the focus keyphrase
- Add this sentence to the beginning of the text (all words from the keyphrase found in the first paragraph within one sentence)  `人によって心地よく感じるポイントは異なりますが、自然の中で本来あるべき場所に、明るく爽やかな森の中を歩く時間は、それだけで心と体を癒してくれるものです。 `
- Confirm that the Keyphrase in first paragraph assessment returns a green bullet.
* Replace the previous sentence with this paragraph (all words from the keyphrase found in the first paragraph in separate sentences): `自然とは、人為によってではなく、おのずから存在しているもの。歩くさわやかな森の中で時間が速くなります。`
- Confirm that the Keyphrase in the first paragraph assessment returns an orange bullet.
* Replace the previous paragraph with this paragraph (one word from the keyphrase found in the first paragraph): `自然とは、人為によってではなく、おのずから存在しているもの。`
- Confirm that the Keyphrase in the first paragraph assessment returns a red bullet.
* Replace the previous paragraph with this sentence (no words from the keyphrase found in the first paragraph): `コンピューターを見つけるのは難しい。`
- Confirm that the Keyphrase in the first paragraph assessment returns a red bullet.

* Disable the `JAPANESE_SUPPORT` by adding this  `define( 'YOAST_SEO_JAPANESE_SUPPORT', false );` in `basic-wordpress-config.php` file
* Add this sentence to the beginning of the text (all words from the keyphrase found in the first paragraph within one sentence), which would normally return a green bullet: `人によって心地よく感じるポイントは異なりますが、自然の中で本来あるべき場所に、明るく爽やかな森の中を歩く時間は、それだけで心と体を癒してくれるものです。`
* Save your changes and reload your page
* Confirm that the Keyphrase  in first paragraph assessment returns a red bullet.


### Test instructions for QA when the code is in the RC



QA can test this PR by following these steps:

This PR can be acceptance tested by following these steps:
- Enable the `JAPANESE_SUPPORT` feature by adding `define( 'YOAST_SEO_JAPANESE_SUPPORT', true );` in `basic-wordpress-config.php` file
- Set your site to Japanese` 日本語`
- Create a post
- Set `自然の中を歩く` as the focus keyphrase
- Add this sentence to the beginning of the text (all words from the keyphrase found in the first paragraph within one sentence)  `人によって心地よく感じるポイントは異なりますが、自然の中で本来あるべき場所に、明るく爽やかな森の中を歩く時間は、それだけで心と体を癒してくれるものです。 `
- Confirm that the Keyphrase in first paragraph assessment returns a green bullet.
* Replace the previous sentence with this paragraph (all words from the keyphrase found in the first paragraph in separate sentences): `自然とは、人為によってではなく、おのずから存在しているもの。歩くさわやかな森の中で時間が速くなります。`
- Confirm that the Keyphrase in the first paragraph assessment returns a orange bullet.
* Replace the previous paragraph with this paragraph (one word from the keyphrase found in the first paragraph): `自然とは、人為によってではなく、おのずから存在しているもの。`
- Confirm that the Keyphrase in the first paragraph assessment returns a red bullet.
* Replace the previous paragraph with this sentence (no words from the keyphrase found in the first paragraph): `コンピューターを見つけるのは難しい。`
- Confirm that the Keyphrase in the first paragraph assessment returns a red bullet.


* Disable the `JAPANESE_SUPPORT` by adding this  `define( 'YOAST_SEO_JAPANESE_SUPPORT', false );` in `basic-wordpress-config.php` file
* Add this sentence to the beginning of the text (all words from the keyphrase found in the first paragraph within one sentence), which would normally return a green bullet: `人によって心地よく感じるポイントは異なりますが、自然の中で本来あるべき場所に、明るく爽やかな森の中を歩く時間は、それだけで心と体を癒してくれるものです。`
* Save your changes and reload your page
* Confirm that the Keyphrase  in first paragraph assessment returns a red bullet.

## Impact check

This PR affects the following parts of the plugin, which may require extra testing:

*

## UI changes


## Other environments


## Documentation


## Quality assurance


Fixes #https://yoast.atlassian.net/jira/software/c/projects/LINGO/boards/99?modal=detail&selectedIssue=LINGO-1088
